make/envoy-gateway.mk

```diff
@@ -0,0 +1,42 @@
+
+##@ Envoy Gateway
+
+## Targets to help install and configure EG
+
+EG_CONFIG_DIR = config/dependencies/envoy-gateway
+EG_NAMESPACE = envoy-gateway-system
+
+# egctl tool
+EGCTL=$(PROJECT_PATH)/bin/egctl
+EGCTL_VERSION ?= v1.0.0
+OS = linux
```

Comment:
Should we get this from the system instead? `$(uname -s)` or something?

Reply:
added automatic detection of OS and ARCH. EG has a custom bash script to do egctl installation: `https://gateway.envoyproxy.io/get-egctl.sh`. Unfortunately, currently it has some limitations, like it requires `sudo`. So, until they enhance their script, we can use our own.